pkg/media/media_data_parser.go

```diff
@@ -113,15 +113,22 @@ func ParseSegmentMediaData(ctx context.Context, mp4bs []byte) (*model.SegmentMed
 	if err != nil {
 		return nil, fmt.Errorf("error creating SegmentMetadata pipeline: %w", err)
 	}
-	demux.Connect("pad-added", onPadAdded)
+	_, err = demux.Connect("pad-added", onPadAdded)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting pad-add: %w", err)
+	}
 
 	go func() {
-		HandleBusMessages(ctx, pipeline)
+		if err := HandleBusMessages(ctx, pipeline); err != nil {
+			log.Log(ctx, "error handling bus messages", "error", err)
+		}
 		cancel()
 	}()
 
 	// Start the pipeline
-	pipeline.SetState(gst.StatePlaying)
+	if err := pipeline.BlockSetState(gst.StatePlaying); err != nil {
```

Comment:
I'm not 100% sure that our usages of `SetState` vs `BlockSetState` are correct but let's not muck with them without a reason
```suggestion
	if err := pipeline.SetState(gst.StatePlaying); err != nil {
```

Reply:
typed the wrong thing here my bad :9